src/ldp/permissions/SparqlPatchPermissionsExtractor.ts

```diff
@@ -0,0 +1,58 @@
+import { Algebra } from 'sparqlalgebrajs';
+import { Operation } from '../operations/Operation';
+import { PermissionSet } from './PermissionSet';
+import { PermissionsExtractor } from './PermissionsExtractor';
+import { Representation } from '../representation/Representation';
+import { SparqlUpdatePatch } from '../http/SparqlUpdatePatch';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+
+/**
+ * Generates permissions for a SPARQL DELETE/INSERT PATCH.
+ * Updates with only an INSERT can be done with just append permissions,
+ * while DELETEs require write permissions as well.
+ */
+export class SparqlPatchPermissionsExtractor extends PermissionsExtractor {
+  public async canHandle(input: Operation): Promise<void> {
+    if (input.method !== 'PATCH') {
+      throw new UnsupportedHttpError('Only PATCH operations are supported.');
+    }
+    if (!input.body) {
+      throw new UnsupportedHttpError('PATCH body is required to determine permissions.');
+    }
+    if (!this.isSparql(input.body)) {
+      throw new UnsupportedHttpError('Only SPARQL update PATCHes are supported.');
+    }
+    if (!this.isDeleteInsert(input.body.algebra)) {
+      throw new UnsupportedHttpError('Only DELETE/INSERT SPARQL update operations are supported.');
+    }
+  }
+
+  public async handle(input: Operation): Promise<PermissionSet> {
+    if (!input.body || !this.isSparql(input.body) || !this.isDeleteInsert(input.body.algebra)) {
+      throw new UnsupportedHttpError('A SPARQL DELETE/INSERT body is required.');
+    }
+    const result = {
+      read: false,
+      append: this.needsAppend(input.body.algebra),
+      write: this.needsWrite(input.body.algebra),
+    };
+    result.append = result.append || result.write;
+    return result;
```

Comment:
Alternatively (just code-style wise, also holds for other checker above):

```suggestion
    const read = false,
          write = this.needsWrite(input.body.algebra),
          append = write || this.needsAppend(input.body.algebra),
    return { read, write, append };
```